kernel/src/process/process_vm/heap.rs

```diff
@@ -60,53 +60,76 @@ impl Heap {
         Ok(())
     }
 
-    pub fn brk(&self, new_heap_end: Option<Vaddr>, ctx: &Context) -> Result<Vaddr> {
+    /// Returns the current program break.
+    pub fn program_break(&self) -> Vaddr {
+        self.current_program_break.load(Ordering::Relaxed)
+    }
+
+    /// Sets the program break to `new_heap_end`.
+    /// Returns the new program break on success, or the current program break on failure.
+    ///
+    /// # Errors
+    /// Returns `Err(current_program_break)` if setting the program break fails. This behavior
+    /// is consistent with the Linux `brk` syscall.
```

Comment:
Style:
```suggestion
    /// Sets the program break to `new_heap_end`.
    ///
    /// Returns the new program break on success, or the current program break on failure.
    /// This behavior is consistent with the Linux `brk` syscall.
```

 - The first paragraph in the doc comments should contain a summary sentence. Keep it short.
 - If you need to write a `# Errors` section, insert a blank line below:
```rust
/// # Errors
///
/// The error section...
```
 - But the error section just repeats information that we've already covered. I don't think it adds much value.